### ISSUE TITLE
SC4-1-2-aria-hidden-focus: Possible false positives

### ISSUE THREAD

**Issue by kasperisager** [auto-wcag/auto-wcag#281]
For `SC4-1-2-aria-hidden-focus`, we require that subtrees that are marked as `aria-hidden=true` do not contain focusable descendants. It did however strike me that on at least one occasion, I've implemented an interaction for a component that didn't serve users of AT well and was therefore hidden using `aria-hidden=true`. An alternative to the interaction was still available so functionality wasn't lost. A case like this would however be a failure of `SC4-1-2-aria-hidden-focus` and I'm not entirely convinced that it should.

**Comment by WilcoFiers** [comment]
I did a little more testing with this. Focusable items inside aria-hidden=true don't get announced, and their accessible name isn't computed, at least not in Chrome and Firefox. I've asked for someone to do this test on Windows as well, I'll update when I know more. I don't think there's anything in spec that suggests the accessible name of these should be computed either.

A focusable element without an accessible name is a pretty straight forward failure of 4.1.2 IMO.

**Comment by kasperisager** [comment]
If that is indeed the case then I don't have any issues with this rule as-is 👍

**Comment by WilcoFiers** [comment]
Yeah, got the results. JAWS seems pretty buggy around focusable links that aren't in the accessibility tree. NVDA seems surprisingly fine with it. I'd still say that, especially because this isn't spect out in any way, we can safely call this a fail. Feel free to close the issue if you agree.
